src/configuration.ts

```diff
@@ -3,29 +3,29 @@ import { ExtensionId } from './constants';
 import { ExtensionContainer } from './container';
 
 export class Configuration {
-	public static configure (context: ExtensionContext) {
+	public static configure (context: ExtensionContext): void {
 		context.subscriptions.push(
-			workspace.onDidChangeConfiguration(configuration.onConfigurationChanged, configuration)
+			workspace.onDidChangeConfiguration(configuration.onConfigurationChanged, configuration) // eslint-disable-line @typescript-eslint/no-use-before-define
 		);
 	}
 
-	public get<T> (section?: string, resource?: Uri, defaultValue?: T) {
+	public get<T> (section?: string, resource?: Uri, defaultValue?: T): T {
 		return defaultValue === undefined
 			? workspace
-					.getConfiguration(section === undefined ? undefined : ExtensionId, resource!)
-					.get<T>(section === undefined ? ExtensionId : section)!
+				.getConfiguration(section === undefined ? undefined : ExtensionId, resource!)
+				.get<T>(section === undefined ? ExtensionId : section)!
 			: workspace
-					.getConfiguration(section === undefined ? undefined : ExtensionId, resource!)
-					.get<T>(section === undefined ? ExtensionId : section, defaultValue)!;
+				.getConfiguration(section === undefined ? undefined : ExtensionId, resource!)
+				.get<T>(section === undefined ? ExtensionId : section, defaultValue)!;
 	}
 
-	public update (section: string, value: any, target: ConfigurationTarget, resourece?: Uri) {
+	public update (section: string, value: any, target: ConfigurationTarget, resourece?: Uri): Thenable<void> {
 		return workspace
 			.getConfiguration(ExtensionId, target === ConfigurationTarget.Global ? undefined : resourece!)
 			.update(section, value, target);
 	}
 
-	private onConfigurationChanged (e: ConfigurationChangeEvent) {
+	private onConfigurationChanged (e: ConfigurationChangeEvent): void {
```

Comment:
<!--
  0 failure: 
  1 warning:  src/configuration...
  
  
  DangerID: danger-id-Danger;
  File: src/configuration.ts;
  Line: 28;
-->

- :warning: src/configuration.ts line 28 – 'e' is defined but never used. (@typescript-eslint/no-unused-vars)